js/test/Exchange/test.sharedMethods.js

```diff
@@ -0,0 +1,206 @@
+'use strict'
+
+const assert = require ('assert');
+const Precise = require ('../../base/Precise');
+
+function logTemplate (exchange, method, entry) {
+    return ' <<< ' + exchange.id + ' ' + method + ' ::: ' + exchange.json (entry) + ' >>> ';
+}
+
+function assertStructureKeys (exchange, method, entry, format, emptyNotAllowedFor = []) {
+    // define common log text
+    const logText = logTemplate (exchange, method, entry);
+    // ensure item is not null/undefined/unset
+    assert (entry, 'item is null/undefined' + logText);
+    // get all expected & predefined keys for this specific item and ensure thos ekeys exist in parsed structure
+    if (Array.isArray (format)) {
+        assert (Array.isArray (entry), 'entry is not an array' + logText);
+        const realLength = entry.length;
+        const expectedLength = format.length;
+        assert (realLength === expectedLength, 'entry length is not equal to expected length of ' + expectedLength.toString () + logText);
+        for (let i = 0; i < format.length; i++) {
+            const is_in_array = exchange.inArray (i, emptyNotAllowedFor);
+            if (is_in_array) {
+                assert ((entry[i] !== undefined), i.toString () + ' index is undefined, but is is was expected to be set' + logText);
+            }
+            // because of other langs, this is needed for arrays
+            try {
+                assert (areSameTypes (entry, i, format), i.toString () + ' index does not have an expected type ' + logText);
+            } catch (ex) {
+                assert (false, i.toString () + ' index is missing from structure' + logText);
```

Comment:
We can't assert False because this is not "valid" python as the linter suggests "Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError()."